src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -442,9 +571,23 @@ CHIP_ERROR ChipDnssdResolve(DnssdService * service, chip::Inet::InterfaceId inte
     VerifyOrReturnError(service != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrReturnError(IsSupportedProtocol(service->mProtocol), CHIP_ERROR_INVALID_ARGUMENT);
 
-    auto regtype     = GetFullType(service);
-    auto interfaceId = GetInterfaceId(interface);
-    return Resolve(context, callback, interfaceId, service->mAddressType, regtype.c_str(), service->mName);
+    auto regtype        = GetFullType(service);
+    auto interfaceId    = GetInterfaceId(interface);
+    const char * domain = nullptr;
+
+    if (BrowseContext::sContextDispatchingSuccess != nullptr)
+    {
+        for (size_t i = 0; i < BrowseContext::sDispatchedServices->size(); ++i)
+        {
+            if (service == &BrowseContext::sDispatchedServices->at(i))
```

Comment:
```suggestion
            if (service == &BrowseContext::sDispatchedServices[i])
```

Reply:
This was not compiling. I can do a pointer->at or a &(*[value[i])

../../examples/chip-tool/third_party/connectedhomeip/src/platform/Darwin/DnssdImpl.cpp:582:25: error: comparison of distinct pointer types ('DnssdService *' and 'std::vector<DnssdService> *')
            if (service == &BrowseContext::sDispatchedServices[i])
                ~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[401/761] objc++ obj/third_party/connectedhomeip/src/platform/Darwin/Darwin.BleConnectionDelegateImpl.mm.o

Reply:
Oh, because sDispatchedServices is a pointer to the vector, right.  OK, `at()` is probably the best answer then.